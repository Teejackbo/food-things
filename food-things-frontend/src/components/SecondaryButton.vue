<template>
  <button v-bind="attributes" :class="type" v-on="listeners"><slot /></button>
</template>

<script>
export default {
  name: 'SecondaryButton',

  props: {
    type: {
      type: String,
      required: false,
    },
  },

  computed: {
    listeners() {
      // eslint-disable-next-line
      return this.$listeners
    },

    attributes() {
      return this.$attrs
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../style/variables';

button {
  background: $grey-lightest;
  border: 0;
  border-radius: 7px;
  color: $grey-darker;
  display: block;
  font-family: $font-heading;
  font-size: 15px;
  font-weight: 600;
  letter-spacing: 0.5px;
  margin-bottom: 1em;
  padding: 0.4em 1em;
  text-transform: uppercase;
  transition: background 0.2s, transform 0.2s;

  &:hover {
    background: darken($grey-lightest, 0.5%);
    transform: translateY(-1px);
  }
}

.error {
  background: $red-lighter;
  color: $red-darkest;

  &:hover {
    background: darken($red-lighter, 0.5%);
  }
}

.success {
  background: $primary-lighter;
  color: $primary-darkest;

  &:hover {
    background: darken($primary-lighter, 3%);
  }
}

.warning {
  background: $yellow-lighter;
  color: $yellow-darkest;

  &:hover {
    background: darken($yellow-lighter, 2%);
  }
}
</style>
