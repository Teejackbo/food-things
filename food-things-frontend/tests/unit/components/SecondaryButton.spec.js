import { shallowMount } from '@vue/test-utils'
import SecondaryButton from '@/components/SecondaryButton.vue'

describe('SecondaryButton', () => {
  it('Should render a button element.', () => {
    const wrapper = shallowMount(SecondaryButton)

    expect(wrapper.contains('button')).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should render a button element with the class of the type prop.', () => {
    const wrapper = shallowMount(SecondaryButton, {
      propsData: {
        type: 'error',
      },
    })

    expect(wrapper.classes()).toContain('error')
    expect(wrapper.element).toMatchSnapshot()
  })
})
