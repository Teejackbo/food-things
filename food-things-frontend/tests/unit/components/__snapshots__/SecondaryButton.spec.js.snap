// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton Should render a button element with the class of the type prop. 1`] = `
<button
  class="error"
/>
`;

exports[`SecondaryButton Should render a button element. 1`] = `<button />`;
